algebra/src/groups/group_vec.rs

```diff
@@ -0,0 +1,245 @@
+    fmt::{Display, Formatter, Result as FmtResult},
+    vec::IntoIter,
+};
+use rand::{
+    distributions::{Distribution, Standard},
+    Rng,
+};
+use core::slice::Iter;
+use num_traits::Zero;
+use serde::*;
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash, CanonicalSerialize, CanonicalDeserialize, Serialize, Deserialize)]
+#[serde(bound(deserialize = "G: Group"))]
+pub struct GroupVec<G: Group> (Vec<G>);
+
+impl<G: Group> GroupVec<G> {
+
+    pub fn new(items: Vec<G>) -> Self { GroupVec(items) }
+
+    pub fn with_capacity(capacity: usize) -> Self {
+        GroupVec(Vec::with_capacity(capacity))
+    }
+
+    pub fn get_vec(&self) -> Vec<G> { self.0.clone() }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn push(&mut self, item: G) {
+        self.0.push(item)
+    }
+
+    pub fn iter(&self) -> Iter<'_, G> {
+        self.0.iter()
+    }
+
+    pub fn into_iter(&self) -> IntoIter<G> {
+        self.0.clone().into_iter()
+    }
+}
+
+impl<G: Group> Distribution<GroupVec<G>> for Standard {
+    #[inline]
+    fn sample<R: Rng + ?Sized>(&self, _rng: &mut R) -> GroupVec<G> {
+        unimplemented!()
+    }
+}
+
+impl<G: Group> Index<usize> for GroupVec<G> {
+    type Output = G;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.0[index]
+    }
+}
+
+impl<G: Group> Default for GroupVec<G> {
+    #[inline]
+    fn default() -> Self {
+        Self::zero()
+    }
+}
+
+impl<G: Group> FromBytes for GroupVec<G> {
+    #[inline]
+    fn read<R: Read>(mut reader: R) -> IoResult<Self> {
+        Ok(GroupVec(CanonicalDeserialize::deserialize_unchecked(&mut reader)
+            .map_err(|e| IoError::new(ErrorKind::Other, format!{"{:?}", e}))?
+        ))
+    }
+}
+
+impl<G: Group> ToBytes for GroupVec<G> {
+    #[inline]
+    fn write<W: Write>(&self, mut writer: W) -> IoResult<()> {
+        CanonicalSerialize::serialize(&self.0, &mut writer)
+            .map_err(|e| IoError::new(ErrorKind::Other, format!{"{:?}", e}))
+    }
+}
+
+impl<G: Group> FromBytesChecked for GroupVec<G> {
+    fn read_checked<R: Read>(mut reader: R) -> IoResult<Self> {
+        Ok(GroupVec(CanonicalDeserialize::deserialize(&mut reader)
+            .map_err(|e| IoError::new(ErrorKind::Other, format!{"{:?}", e}))?
+        ))
+    }
+}
+
+impl<G: Group> SemanticallyValid for GroupVec<G> {
+    fn is_valid(&self) -> bool {
+        for item in self.0.iter() {
+            if !item.is_valid() {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+impl<G: Group> Display for GroupVec<G> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        for (i, item) in self.0.iter().enumerate() {
+            writeln!(f, "[{}]: {}", i, item)?;
+        }
+        Ok(())
+    }
+}
+
+impl<G: Group> Zero for GroupVec<G> {
+    #[inline]
+    fn zero() -> Self {
+        GroupVec(vec![])
+    }
+
+    #[inline]
+    fn is_zero(&self) -> bool {
+        self.0.len() == 0
+    }
```

Comment:
Don't we have different representations of zero here, too? (As we have for polynomials.)

Reply:
Yes I think that the current implementation of `is_zero()` produces "false negatives".
I would propose the following implementation instead, which takes into account trailing zeros.
```suggestion
    fn is_zero(&self) -> bool {
        self.0.iter().all(|&el| el.is_zero())
    }
```

Reply:
Done

Reply:
Modified like in Polynomial